File: elixir/square-root/lib/square_root.ex
defmodule SquareRoot do
  @doc """
  Calculate the integer square root of a positive integer
  """
  @spec calculate(radicand :: pos_integer) :: pos_integer
  def calculate(radicand) do
    sqrt(radicand, 1, 0)
  end

  # Repeated Subtraction Method
  defp sqrt(radicand, current_odd_number, counter) when radicand < current_odd_number, do: counter

  defp sqrt(radicand, current_odd_number, counter) do
    sqrt(radicand - current_odd_number, current_odd_number + 2, counter + 1)
  end
end

File: java/square-root/src/main/java/SquareRoot.java
public class SquareRoot {
    public int squareRoot(int radicand) {

        // Repeated subtraction method
        int currentOddNumber = 1;
        int counter = 0;

        while (radicand >= currentOddNumber) {
            radicand -= currentOddNumber;
            currentOddNumber += 2;
            counter++;
        }

        return counter;
    }
}
